File: picmaker.c
#include <stdio.h>
#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>
#include <errno.h>
#include <string.h>

int main () {
  int fd = open ("pic.ppm", O_WRONLY|O_CREAT);
  char firstlines[] = "P3 100 100 255";
  write (fd, firstlines, sizeof (firstlines));
  char rgb[3][3];
  int i,j;
  for (i = 0; i < 10; i++) {
    if (i < 3333) {
      itoa (255, rgb[0], sizeof (rgb[0]));
      itoa (random() %256, rgb[1], sizeof (rgb[1]));
      itoa (random() %256, rgb[2], sizeof (rgb[2]));
    }
    else if (i < 6666) {
      itoa (random() %256, rgb[0], sizeof (rgb[0]));
      itoa (255, rgb[1], sizeof (rgb[1]));
      itoa (random() %256, rgb[2], sizeof (rgb[2]));
    }
    else {
      itoa (random() %256, rgb[0], sizeof (rgb[0]));
      itoa (random() %256, rgb[1], sizeof (rgb[1]));
      itoa (255, rgb[2], sizeof (rgb[2]));
    }
    char color [12];
    strcpy (color, rgb[0]);
    strcat (color, rgb[1]);
    strcat (color, rgb[2]);
    write (fd, color, sizeof (color));
  }
  return 0;
}

File: picmaker.py
import random
f = open ("pic.ppm", "w")

f.write ("P3 500 500 255 ")

color = []

# def placecolors ():
#     for i in range (500*500):
#         red = random.randint (1,255)
#         green = random.randint (1,255)
#         blue = random.randint (1,255)
#         f.write (str (red) + " " + str (green) + " " + str (blue) + " ")

colors = []

def placecolors ():
    for y in range (500):
        for x in range (500):
            if pow (y + 250, 2) - pow (x + 250, 2) == 1:
                color = [255, 100, 34]
                colors.append (color)
            elif y % 4 == 0 and x % 4 == 0:
                color = [0, 0, 255]
                colors.append (color)
            else:
                color = [100, 250, 90]
                colors.append (color)

def toFile ():
    for i in range (len (colors)):
        f.write (str (colors[i][0]) + " " + str (colors[i][1]) + str (colors[i][2]) + " ")

placecolors ()
toFile ()
